src/django/api/processing.py

```diff
@@ -187,11 +187,12 @@ def parse_facility_list_item(item):
 
         parser = CsvRowSectorProductTypeParser(fields, values)
         item.sector = parser.sectors
-        if 'product_type' in fields:
-            values[fields.index('product_type')] = parser.product_types
-        else:
-            fields.append('product_type')
-            values.append(parser.product_types)
+        if len(parser.product_types) > 0:
+            if 'product_type' in fields:
+                values[fields.index('product_type')] = parser.product_types
+            else:
+                fields.append('product_type')
+                values.append(parser.product_types)
```

Comment:
What about when a user enters only valid sectors under `product_types`? In this case we need to delete the product type field so those values are not also added as product types. Here's a test:

```python
    def test_only_valid_sectors_as_product_type(self):
        facility_list = FacilityList.objects.create(
            header="sector_product_type,address,country,name"
        )
        source = Source.objects.create(
            source_type=Source.LIST, facility_list=facility_list,
            contributor=self.contributor
        )
        item = FacilityListItem.objects.create(
            row_index=1,
            sector=[],
            raw_data="Apparel,1234 main st,ChInA,Shirts!",
            source=source
        )
        parse_facility_list_item(item)
        self.assert_successful_parse_results(item)
        self.assertEqual(['Apparel'], item.sector)
        self.assertEqual(0, ExtendedField.objects.all().count())
```